src/components/WebComponent/Project/WebComponentProject.js

```diff
@@ -42,7 +42,7 @@ const WebComponentProject = () => {
       webComponent.dispatchEvent(customEvent);
     }, 2000);
     setTimeoutId(id);
-  }, [project]);
+  }, [project, timeoutId, webComponent]);
```

Comment:
Are you sure this doesn't make the browser go crazy? The reason the `timeoutId` was left out initially was because it was causing constant rerenders (which was throwing up a warning in the console)

Reply:
Will run it locally and see the console, thanks a lot for flagging it up

Reply:
@loiswells97 I wrapped the method to clear any existing timeout when the dependencies change. Could you please check the console if there are any issues? I couldn't replicate any console issue

Reply:
Okay, I took a look and I'm not sure the custom event is still being triggered. For example, if you go to the [web component test page](https://staging-editor.raspberrypi.org/branches/issues_668_Fix_useEffect_web_component_warnings/web-component.html) it should print the code to the console 2 seconds after it's changed (in response to the `codeChanged` event), but that isn't happening anymore.

Reply:
@loiswells97 I removed the reset of timeoutId. I wonder if adding a timeout clearing function (clearing interval) might be a good idea to include the timeoutId within the useEffect() dependencies and also make sure it's not causing any console issues.

Reply:
On your point, I'm not sure, I feel like my understanding of the timeout stuff is sketchy at best